### PR TITLE
feat: validate development_stage_ontology_term_id for Zebrafish

### PR DESCRIPTION
**Pull request by nayib-jose-gloria**
## Reason for Change

- #1052 & #1053 
- see https://github.com/chanzuckerberg/single-cell-curation/blob/main/schema/drafts/5.2.1-experimental.md#development_stage_ontology_term_id

## Changes

- use 5.2.1-experimental COG/ontology parser 
- add validation requirements for development_stage_ontology_term_id for Zebrafish

## Testing

- unit tests

### CODE REVIEW

File: cellxgene_schema_cli/cellxgene_schema/schema_definitions/schema_definition.yaml [hunk 1]
```diff
@@ -316,9 +316,28 @@ components:
...
+            error_message_suffix: >-
+              When 'organism_ontology_term_id' is 'NCBITaxon:7955' (Danio rerio),
+              'development_stage_ontology_term_id' MUST be the most accurate descendant of 'ZFS:0000001' and it
+              MUST NOT be 'ZFS:0000000' for Unknown. The str 'unknown' is acceptable.
+            type: curie
+            curie_constraints:
+              ontologies:
+                - ZFS
+              allowed:
+                ancestors:
```
Comment (nayib-jose-gloria) [inline cellxgene_schema_cli/cellxgene_schema/schema_definitions/schema_definition.yaml:330]: It is a little confusing, but we use the ['allowed']['ancestors'] key to mean: we only allow terms that have one of the following terms as an ancestor

(NOT we allow ancestors of the one of the following terms)

**Review by joyceyan — APPROVED** [review]

File: .github/workflows/push_tests.yml [hunk 2]
```diff
@@ -4,7 +4,7 @@ on:
   push:
     branches: main
   pull_request:
-    branches: "*"
+    branches: "**"
```
[the remark this answers is not in the document]
Comment (nayib-jose-gloria) [reply .github/workflows/push_tests.yml:7]: I think its a GHA syntax thing, it was always meant to run the unit tests on all PRs but the previous version wasn't working when i initially opened this one

File: cellxgene_schema_cli/setup.py [hunk 3]
```diff
@@ -5,7 +5,7 @@
 
 setup(
     name="cellxgene-schema",
-    version="5.2.0",
+    version="5.2.1-alpha",
```
[the remark this answers is not in the document]
Comment (nayib-jose-gloria) [reply cellxgene_schema_cli/setup.py:8]: the pypi build doesn't allow `5.2.1-experimental`, it only allows rc, alpha or beta after a '-'. However it does allow 5.2.1+experimental. Should we change to that? It uses [PEP 440](https://peps.python.org/pep-0440/) to validate versions

[the remark this answers is not in the document]
Comment (joyceyan) [reply cellxgene_schema_cli/setup.py]: maybe we should just call this `5.2.1-alpha`? iirc semvar doesn't allow the `+` char

**Pull request merged**